packages/astro/src/core/csp/common.ts

```diff
@@ -0,0 +1,40 @@
+import type { AstroConfig } from '../../types/public/index.js';
+import type { BuildInternals } from '../build/internal.js';
+import crypto from 'node:crypto';
+import type { AstroSettings } from '../../types/astro.js';
+
+export function shouldTrackCspHashes(config: AstroConfig): boolean {
+	return config.experimental?.csp === true;
+}
+
+export function trackStyleHashes(internals: BuildInternals): string[] {
+	const clientStyleHashes: string[] = [];
+	for (const [_, page] of internals.pagesByViteID.entries()) {
+		for (const style of page.styles) {
+			if (style.sheet.type === 'inline') {
+				clientStyleHashes.push(
+					crypto.createHash('sha256').update(style.sheet.content).digest('base64'),
```

Comment:
Should the code used to generate hashes be abstracted/extracted?

Reply:
I don't understand what you mean. Can you explain?

Reply:
Sorry, I mean that `crypto.createHash('sha256').update(CONTENT).digest('base64')` is duplicated in 3 places IIRC so maybe that could be extracted. Non blocking!

Reply:
Oh, I see. I'll do it, thank you